src/sensai/vector_model.py

```diff
@@ -133,27 +125,119 @@ def withName(self, name: str):
         self.setName(name)
         return self
 
+    def getInputTransformer(self, cls: Type[DataFrameTransformer]):
+        for it in self._inputTransformerChain.dataFrameTransformers:
+            if isinstance(it, cls):
+                return it
+        return None
+
+    def setName(self, name):
+        self._name = name
+
+    def getName(self):
+        if self._name is None:
+            return "unnamed-%s-%x" % (self.__class__.__name__, id(self))
+        return self._name
+
+    # TODO: why do we need the getter and the setter? We only have them for fgens, not for the input/output/target transformers
```

Comment:
@opcode81 What do you think of this? I would be in favor of removing the getter and setter, or do you see good use cases for them?

Reply:
What would be the benefit of removing them?
The getter was used in at least one project, and the setter, while functionally redundant, makes sense for post-construction use, where writing model.withFG(fg) would feel weird.

Reply:
It just felt strange that this exists only for fgen. If we want to allow getter and setter, why not for all other components as well?

Reply:
The reason is that sensAI features were added only when needed.
If you think that users could reasonably want these getters/setters, you can add them, but you should have a reasonable use case in mind. If the use case is entirely "hacky", then we shouldn't add them, as users can make do with equally hacky private access. :)

Reply:
That's what I suspected ^^. Ok, let's leave it as is, certainly there is no harm.